File: 169-majority-element/majority-element.go
func majorityElement(nums []int) int {
    m := make(map[int]int)
    for i := 0; i < len(nums); i++ {
        if _, ok := m[nums[i]]; ok {
            m[nums[i]] += 1
        } else {
            m[nums[i]] = 1
        }

        if m[nums[i]] > len(nums) / 2 {
            return nums[i]
        }
    }

    return nums[0]
}

File: 169-majority-element/majority-element.dart
class Solution {
  int majorityElement(List<int> nums) {
    final mp = <int, int>{};
    final mj = nums.length ~/ 2;

    for (final e in nums) {
        mp[e] = (mp[e] ?? 0) + 1;
        if (mp[e]! > mj) return e;
    }

    return nums[0];
  }
}